src/tool/subcommands/api_cmd.rs

```diff
@@ -746,6 +746,31 @@ fn websocket_tests() -> Vec<RpcTest> {
     vec![test]
 }
 
+// Sanity check `ApiInfo` and `ApiTestFlags` args
+fn validate_protocols(
+    forest: &ApiInfo,
+    lotus: &ApiInfo,
+    config: &ApiTestFlags,
+) -> anyhow::Result<()> {
+    let a = forest.multiaddr.clone().pop().map(|p| p.tag());
+    let b = lotus.multiaddr.clone().pop().map(|p| p.tag());
+
+    let is_valid = match (a, b) {
```

Comment:
1. Is it fine if either `a` or `b` are `None`? This match will return `true` in such a case.
2. This seems a very complex way to assert inequality of two values (first 4 lines). Can we simplify it?

Reply:
Indeed. Good catch for 1. I will simplify this function and add proper testing.